injector/injector.go

```diff
@@ -93,26 +103,22 @@ func mutate(body []byte, coordAddr string, domainName string, resourceKey string
 		},
 	}
 
-	// get marble type from pod labels
-	marbleType := pod.Labels["marblerun/marbletype"]
-	// allow pod to start if label does not exist, but dont inject any values
-	if len(marbleType) == 0 {
-		admReviewResponse.Response.Allowed = true
-		admReviewResponse.Response.Result = &metav1.Status{
-			Status:  "Success",
-			Message: "Missing [marblerun/marbletype] label, injection skipped",
-		}
-		bytes, err := json.Marshal(admReviewResponse)
-		if err != nil {
-			log.Println("Error: unable to marshal admission response")
-			return nil, errors.New("unable to marshal admission response")
-		}
-		log.Println("Pod is missing [marblerun/marbletype] label, skipping injection")
-		return bytes, nil
+	if pod.Labels == nil {
+		pod.Labels = make(map[string]string)
 	}
 
-	pT := v1.PatchTypeJSONPatch
-	admReviewResponse.Response.PatchType = &pT
+	// get marble type from pod labels
+	marbleType, exists := pod.Labels[labelMarbleType]
+	if !exists {
+		// allow pod to start if label does not exist, but dont inject any values
+		log.Printf("Pod is missing [%s] label, skipping injection", labelMarbleType)
+		return generateResponse(pod, admReviewReq, admReviewResponse, true, fmt.Sprintf("Missing [%s] label, injection skipped", labelMarbleType))
```

Comment:
```suggestion
		msg := fmt.Sprintf("Pod is missing [%s] label, skipping injection", labelMarbleType)
		log.Println(msg)
		return generateResponse(pod, admReviewReq, admReviewResponse, true, msg)
```
This pattern repeats multiple times, please fix them all.